src/novel_applications/defi/clob.md

```diff
@@ -0,0 +1,167 @@
+
+```
+receive( 1 ETH )  # what is in the vault
+send( 2000 DAI, note.sender ) # send to the creator of this note
+```
+
+Whoever wants to take this order needs to consume the above note and, in doing so, execute the script. Like this, we can pretty easily support limit orders, where limits can be updated nearly instantaneously in one direction.
+
+The order book could now list all notes with the same `Tag`.
+    
+![](https://hackmd.io/_uploads/ryr24LsS2.png)
+
+This will serve as our order book - which can be shown in a simple web2 interface/homepage. Now any user can take any order by simply consuming the respective note in a transaction - first come, first serve. 
+
+Notes in that order book must be public. That means everyone sees all the note's data and can compute the nullifier. That means whenever there is a nullifier in the Nullifier DB, the operator of the order book interface can delete the respective note. 
+
+### Limit Orders Can Be Updated for Free by Revealing Parts of the Script
+
+A limit order is a direction to buy or sell an asset for a specific price (limit) or better. We can update those limits by revealing new parts of the note script. The note's script can be partially hidden at the beginning, and whenever a user wants to update a limit order, a new leaf can be revealed in no time. 
+    
+![7mz95q](https://github.com/0xPolygonMiden/examples/assets/35031754/6fa0a333-f6a7-46e5-aca6-fa5206eda427)
+
+Now the user first reveals what A is. Then, what B is, and so on. 
+    
+```
+A = {receive 1 ETH && send 2000 DAI to sender}
+```
+
+If no one takes that order, the user reveals another part of the script.
+
+```
+B = {receive 1 ETH && send 1990 DAI to sender}
+```
+    
+### Market Orders Are Slightly More Complex
+
+A market order is an order to buy or sell an asset immediately. This type of order guarantees that the order will be executed but does not guarantee the execution price. 
+
+A naive implementation would be to create a note having a script that queries the current market price. The script could be like `receive( 1 ETH ), send( current_price, note.sender )`. Whereas `current_price` must come from an oracle but it should be easy to compute given that all notes are public. However, there might be attack vectors for nonliquid markets. 
+
+So now, users can place limit and market orders by creating new notes. Those orders can be updated by simply revealing new parts of the script. Those updates don't need to happen on-chain, but, e.g., in any public forum or directly to the operator, and are therefore for free and can happen in no time. And anybody can take orders by simply consuming the respective notes.
```

Comment:
Sometimes in this text you use `onchain` and sometimes `on-chain`. Let's use always `on-chain`